### DISCUSSION TITLE
Schema Validation - Please Help

### DISCUSSION THREAD

**Discussion by sanathannarayanan** [Cornutum/tcases#175] [Q&A] [answered]
Hi, I have tried out tcases project and really like it. It is very professional, and has great functionalities.

**My Problem Statement:**

- I want to autotmatically generate testcases for custom REST APIs and also execute the test cases with this same tool

**My Progress:**

- I could successfully generate and run 12 testcases.. the validation was done using "status code" by the tool

**What I want:**

- Is there a way to generate and run the testcases, where the tool should validate based on "the response" of API and not just the status code..
- if yes, then where and how can i do it??

**Future Scope:**

- Also, does this tool support giving multiple APIs as inputs in a single go, instead of running it with single API many times??

  
 Can anyone please help me out with this?? @tkruse @kerrykimbrough @callmekungfu

**Comment by kerrykimbrough** [comment, marked as answer]
You're right: Tcases doesn't generate assertions about expected responses to the requests in generated test cases. Tcases doesn't really know anything about what the API does. So there is no way for Tcases to predict what the response to any request should be, much less for requests with randomly-generated inputs.

**Comment by kerrykimbrough** [comment]
> does this tool support giving multiple APIs as inputs in a single go?

Yes, you can do this using Maven to run the Tcases Maven Plugin. Specifically, see the [tcases:api-test](http://www.cornutum.org/tcases/docs/tcases-maven-plugin/api-test-mojo.html) goal. This goal will generate tests for all OpenAPI docs in a directory tree. Various options control the input directory and which API doc files to process.
